analytics/tracker_object.rst

```diff
@@ -0,0 +1,543 @@
+.. highlight:: js
+.. default-domain:: js
+
+Tracker Object Functions
+========================
+
+This document describes all functions available for Tracker object and how to create its instance.
+This enables user to track on multiple trackers at once.
+
+Accessing Tracker Object
+------------------------
+
+To access tracker object instance you must use  ``Piwik.getTracker`` function
+
+.. function:: Piwik.getTracker(trackerUrl, siteId);
+
+    Getter for Analytics Tracker instance.
+
+    :param string trackerUrl: **Required** Url for tracker
+    :param string siteId: **Required** Site Id that will be linked to tracked data.
+    :returns: Analytics Tracker instance
+
+
+
+To access internal instance of the Tracker used for asynchronous tracking you must use  ``Piwik.getAsyncTracker`` function
+
+.. function:: Piwik.getAsyncTracker(trackerUrl, siteId);
+
+    Getter for Analytics Tracker instance.
+
+    :param string trackerUrl: **Required** Url for tracker
+    :param string siteId: **Required** Site Id that will be linked to tracked data.
+    :returns: Analytics Tracker instance
+
+Tracking functions
+------------------
+
+.. function:: trackPageView(customPageTitle);
+
+    Tracks a visit for page that function was run on.
+
+    :param string customPageTitle: **Required** Custom page title, for example ``document.title``
+
+.. function:: trackEvent(category, action, name, value);
+
+    Tracks events that should not trigger on page load, but when user performs an action
+
+    :param string category: **Required** Category of event.
+    :param string action: **Required** Event action, for example ``"link click"``.
+    :param string name: **Optional** Event name, for example ``"Cancel button"``.
+    :param string value: **Optional** Event value.
+
+.. function:: trackGoal(idGoal, customRevenue, customData);
+
+    Manualy tracks goal (conversion).
+
+    :param int/string idGoal: **Required** Id of goal.
+    :param int/float customRevenue: **Optional** Revenue value
+    :param mixed customData: **Optional** Object that can :ref:`Custom dimensions`.
+
+.. todo:: What else can be in customData?
+
+.. function:: trackSiteSearch(keyword, category, resultCount);
+
+    Function that will track internal site searches.
+
+    :param string keyword: **Required** String containing keyword that was searched.
+    :param string/boolean category: **Required** String with category selected in search engine, can set it to false when not used.
+    :param number/boolean searchCount:  **Required** Number of results on the results page, can be set to false when not used.
+
+.. function:: enableHeartBeatTimer(delay);
+
+    When user will enter single page on visit we will assume that total time spent on website was 0 ms.
+Function will measure that time more accurately.
```

Comment:
Probably missing `this` before function. It's somewhat inconsistent, it does not measure, it enables this setting.